### PR TITLE
QHY driver improvements

### PR DESCRIPTION
**Pull request by daleghent**
This PR addresses the points raised in #1245 

The new options window presents thusly:

image

Tooltips are provided for each option that explains its purpose and any model-specific relevancy.

### CODE REVIEW

**Comment by daleghent** [comment]
A little editorializing: Release notes ought to mention that it would be a good idea for QHY guide camera users to do a fresh dark library run for their camera. While I took pains with the defaults to ensure that cameras will produce the same output as before this change, that can't be guaranteed on a model-by-model basis. Plus, a fresh dark library is good hygiene anyhow.

File: src/cam_qhy.cpp [hunk 1]
```diff
@@ -556,6 +900,25 @@ bool Camera_QHY::Capture(int duration, usImage& img, int options, const wxRect&
...
+        {
+            m_offset = m_offsetMin;
+        }
+
+        if (m_offset > m_offsetMax)
+        {
+            m_offset = m_offsetMax;
+        }
+
+        double offset = floor(m_offset / m_offsetStep) * m_offsetStep;
```
Comment (agalasso) [inline src/cam_qhy.cpp:914]: can we add some protection against `m_offsetStep == 0`, either here or back when we call `GetQHYCCDParamMinMaxStep`?

File: src/cam_qhy.cpp [hunk 2]
```diff
@@ -572,14 +935,29 @@ bool Camera_QHY::Capture(int duration, usImage& img, int options, const wxRect&
...
+        {
+            Debug.Write(wxString::Format("QHY: 2000ms delay needed\n"));
+            WorkerThread::MilliSleep(2000);
+        }
     }
-    if (ret == QHYCCD_READ_DIRECTLY)
+    else
     {
-        // Debug.Write("QHYCCD_READ_DIRECTLY\n");
+        Debug.Write(wxString::Format("QHY: %dms delay needed\n", duration));
```
Comment (agalasso) [inline src/cam_qhy.cpp:959]: optional: maybe remove this? may be a little excessive logging this especially for small durations where the logging it self could even take longer than the requested delay

File: src/cam_qhy.h [hunk 3]
```diff
@@ -35,6 +35,9 @@
 #ifndef CAM_QHY_INCLUDED
 #define CAM_QHY_INCLUDED
 
+#define QHYCCD_OFF  0.0
+#define QHYCCD_ON   1.0
+
```
Comment (agalasso) [inline src/cam_qhy.h:40]: let's move these definitions into cam_qhy.cpp as they do not need to be exposed here as part of the pure interface being defined in this file.

**Review by agalasso — COMMENTED** [review]
great stuff, thanks!

File: src/cam_qhy.cpp (hunk 2 above)
Comment (daleghent) [reply src/cam_qhy.cpp:959]: Definitely. I had that there to follow what was there previously as well as to observe how different camera models did their return codes for the exposure. While neat, it's certainly not needed.

File: src/cam_qhy.h (hunk 3 above)
Comment (daleghent) [reply src/cam_qhy.h]: Done

File: src/cam_qhy.cpp [hunk 4]
```diff
@@ -291,6 +584,36 @@ bool Camera_QHY::Connect(const wxString& camId)
         return CamConnectFailed(_("Failed to get gain range"));
     }
 
+    if (GetQHYCCDParamMinMaxStep(m_camhandle, CONTROL_OFFSET, &m_offsetMin, &m_offsetMax, &m_offsetStep) == QHYCCD_SUCCESS)
```
Comment (agalasso) [inline src/cam_qhy.cpp:587]: could we do something like this so if the API call sets m_offsetStep to zero we don't get a divide by zero later on.  Should never happen if the SDK is behaving properly, but prefer not to make that assumption.
```suggestion
    if (GetQHYCCDParamMinMaxStep(m_camhandle, CONTROL_OFFSET, &m_offsetMin, &m_offsetMax, &m_offsetStep) == QHYCCD_SUCCESS && m_offsetStep > 0)
```

**Review by agalasso — COMMENTED** [review]
just one comment.. if we could resolve that then happy to merge it and get it into the next release

File: src/cam_qhy.cpp (hunk 4 above)
Comment (daleghent) [reply src/cam_qhy.cpp]: Hey Andy, I made that change (and now guard against a 0 step in several places) and a few others, as well.

**Comment by daleghent** [comment]
I've collected a small list of folks who are interested in testing this new code. I'm sending an email to this evening with instructions. I'll CC you and Bruce and BCC the interested parties.

**Comment by daleghent** [comment]
I've ran tests of the current code overnight and no memory leaks are apparent.

**Review by agalasso — APPROVED** [review]
great work, thanks!

**Pull request merged**